src/passes/shadowstack.ts

```diff
@@ -162,6 +162,18 @@ function matchTostack(module: Module, expr: ExpressionRef): ExpressionRef {
   return 0;
 }
 
+/** Tests whether a `value` matched by `matchTostack` needs a slot. */
+function needsSlot(module: Module, value: ExpressionRef): bool {
+  switch (_BinaryenExpressionGetId(value)) {
+    // no need to stack null pointers
+    case ExpressionId.Const: return !isConstZero(value);
+    // already kept in another slot
+    case ExpressionId.LocalGet:
+    case ExpressionId.LocalSet: return false; // tee
```

Comment:
`case ExpressionId.LocalTee`?

Reply:
In Binaryen, a `local_tee` is just a `LocalSet` with `.tee = true`.

Reply:
I see. Thanks!